### PR TITLE
Add routine for changing the ldap admin password

### PR DESCRIPTION
**Pull request by pwalczysko**
Taking on board the comment https://github.com/openmicroscopy/management_tools/pull/674#issuecomment-365882289
This adds to the ``ldapmanager`` script the routine for changing the ldap password.

``./ldapmanager ldappwd --new-pass $NEWPASSWORD``

cc @joshmoore - happy to add, rename and explain as necessary

### CODE REVIEW

**Comment by joshmoore** [comment]
Interesting. For a moment, I thought that this was duplicate work with https://github.com/openmicroscopy/apacheds-docker/pull/10 but in fact this is the admin password while gh-10 is for user passwords. We use unify our naming. `passwd` and `adminpasswd`?

**Comment by pwalczysko** [comment]
Will add the travis steps tomorrow.

Maybe I could suggest a common method for the repeating bits here which I borrowed from https://github.com/openmicroscopy/apacheds-docker/pull/10#pullrequestreview-97654454 ?

**Comment by joshmoore** [comment]
Removing `Align the adminpasswd command as per Josh's comment` 4ae4801 to simplify the combining.

**Comment by joshmoore** [comment]
@pwalczysko : please see the new proposal to combine our two PRs. `adminpasswd` is now `passwd --admin`.

File: bin/ldapmanager [hunk 1]
```diff
@@ -70,6 +92,41 @@ def main():
...
 
+    # Passwd ################################################################
+    passwd = subparsers.add_parser(
+        "passwd", help="reset or add passwords for users",
+        description="""
+Each user can have one or more passwords. This command allows either replacing
+all current passwords (default) or appending passwords with --append. Passwords
+can be set for one user at a time. If no option is chosen, then the current user
+(--bind-user) will be used. --admin will use the value of $LDAP_ADMIN_DN
+(default: uid=admin,ou=system) while --login will change the password of a user
```
Comment (pwalczysko) [inline bin/ldapmanager]: ``--login will change the password....`` -> ``--login  can be used to specify a user in ou=Users``

File: bin/ldapmanager [hunk 2]
```diff
@@ -22,13 +30,12 @@ Example
     ./ldapmanager user u2
     ./ldapmanager user u3
 
-    ./ldapmanager passwd u1  # Requires input
-    ./ldapmanager passwd u1 --password secret
-    echo secret | ./ldapmanager password u1 -f-
+    ./ldapmanager passwd --login u1  # Requires input
```
Comment (pwalczysko) [inline bin/ldapmanager]: Exactly this line fails for me with 

```
./ldapmanager --bind-pass $MYBINDPWD passwd --login mzeman
Password: 
Invalid Dn given : uid=mzeman,ou=Users,None (0x75 0x69 0x64 0x3D 0x6D 0x7A 0x65 0x6D 0x61 0x6E 0x2C 0x6F 0x75 0x3D 0x55 0x73 0x65 0x72 0x73 0x2C 0x4E 0x6F 0x6E 0x65 ) is invalid
Invalid DN syntax
```

Comment (pwalczysko) [reply bin/ldapmanager]: The ``--admin`` flag worked in all tests as expected, but see https://github.com/openmicroscopy/apacheds-docker/pull/9#discussion_r169443150 - is the dn creation in https://github.com/openmicroscopy/apacheds-docker/pull/9/files#diff-278610078cfba3d472f5f3bcbd71ff58R337 at fault ?

**Comment by pwalczysko** [comment]
The ``--admin`` flag worked in all tests as expected, but see https://github.com/openmicroscopy/apacheds-docker/pull/9#discussion_r169443150 - is the dn creation in https://github.com/openmicroscopy/apacheds-docker/pull/9/files#diff-278610078cfba3d472f5f3bcbd71ff58R337 at fault ?

File: bin/ldapmanager (hunk 2 above)
Comment (joshmoore) [reply bin/ldapmanager]: Hmmm.... yes, I have a chicken and the egg problem. Working on a solution

**Comment by joshmoore** [comment]
Attempted fix pushed, @pwalczysko.

**Comment by pwalczysko** [comment]
@joshmoore Thank you. Yes, the last commit fixed the targeted bug. But it either introduced another or the second bug (below) was there already before, cannot tell:

```
./ldapmanager --bind-pass $MYBINDPWD passwd
Password: 
NO_SUCH_OBJECT: failed for MessageType : MODIFY_REQUEST
Message ID : 2
    Modify Request
        Object : 'uid=None,ou=Users,dc=openmicroscopy,dc=org'
            Modification[0]
                Operation :  replace
                Modification
userPassword: 0x6D 0x6F 0x72 0x61 0x76 0x61 org.apache.directory.api.ldap.model.message.ModifyRequestImpl@5669315f: Attempt to lookup non-existant entry: uid=None,ou=Users,dc=openmicroscopy,dc=org
No such object
```

To be honest, I would consider it a good thing to remove the ``passwd`` with no flags possibility alltogether.  Neither me nor my ldap understand who is the ``current user`` in https://github.com/pwalczysko/apacheds-docker/blob/575f3a5ff480e1391c83aae8bf3b8ecd37a39c75/bin/ldapmanager#L38  :)

**Comment by joshmoore** [comment]
> To be honest, I would consider it a good thing to remove the passwd with no flags possibility alltogether.

Can do, but note that is also the behavior of Unix' `passwd` as well as OMERO's `bin/omero user password`.

**Comment by pwalczysko** [comment]
I would gladly accept this plugin not to be ``perfect-perfect``.  Dropping this option and finalizing the efforts here seems to me a very good solution.

**Comment by joshmoore** [comment]
Pushed as suggested.

**Comment by pwalczysko** [comment]
Great, thank you. 

This works as expected.

Ready to merge FMPOV.

**Comment by joshmoore** [comment]
Propose tag as `0.5.0`?

**Comment by pwalczysko** [comment]
>Propose tag as 0.5.0?

Fine by me.

**Pull request merged**